### PR TITLE
Update image_dataset.py

### LINKED ISSUE #49829: image_dataset_from_directory returns dtype 'uint8' Dataset if interpolation='nearest'
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image_dataset_from_directory

### Clear description
tf.keras.preprocessing.image_dataset_from_directory returns dtype 'uint8' Dataset if interpolation='nearest'
however function docstring reports misleading:

### Returns defined

Returns:    A `tf.data.Dataset` object.      - If `label_mode` is None, it yields `float32` tensors of shape        `(batch_size, image_size[0], image_size[1], num_channels)`,        encoding images (see below for rules regarding `num_channels`).      - Otherwise, it yields a tuple `(images, labels)`, where `images`        has shape `(batch_size, image_size[0], image_size[1], num_channels)`,        and `labels` follows the format described below.

[...]
## Description of issue (what needs changing):

return dtype 'uint8' for interpolation='nearest should be specified, such as:

Returns:    A `tf.data.Dataset` object.      - If `label_mode` is None, it yields `float32` tensors of shape        `(batch_size, image_size[0], image_size[1], num_channels)`,        encoding images (see below for rules regarding `num_channels`).     
 - If `label_mode` is None and interpolation is 'bilinear' it yields `uint8`.
 - Otherwise, it yields a tuple `(images, labels)`, where `images`        has shape `(batch_size, image_size[0], image_size[1], num_channels)`,        and `labels` follows the format described below.

[...]

### PR DESCRIPTION
**Pull request by Saduf2019**
Updated return values for "tf.keras.preprocessing.image_dataset_from_directory"

Fixes #49829

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
For these doc changes we should also have a doctest example to accompany them.

Since this seems `tf.data`, adding Jiri to review.

**Comment by jsimsa** [comment]
This is Keras. Adding @fchollet for triage.

**Comment by fchollet** [comment]
Thanks for the PR. We no longer develop Keras via the TensorFlow GitHub, so please reopen the PR at https://github.com/keras-team/keras
